### PR TITLE
add me-tokens to dropdown

### PR DESCRIPTION
**Pull request by Seroxdesign**
## Overview

**What features/fixes does this PR include?**

Integrates Metokens into the MetaOS custom blocks

**Please provide the GitHub issue number**

## Follow up Improvement Ideas

- [ ] please list any improvement/ideas

## Implementation

**Describe technical (nontrivial / non-obvious) parts of your code**

**Side effects**

## Assets

[Include screenshots/videos if it makes reviewing easier.]

### CODE REVIEW

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (alalonde) [inline packages/web/components/Player/Section/PlayerMeToken.tsx:118]: All these functions should either be wrapped in `useCallback` or moved outside the render function. As-is they are recreated on every single rendering pass

File: packages/web/utils/orbis.d.ts
Comment (alalonde) [inline packages/web/utils/orbis.d.ts:1]: Can you put this with the other external types in `packages/@types`?

File: packages/web/utils/meTokens.tsx
Comment (alalonde) [inline packages/web/utils/meTokens.tsx:1]: If this doesn't contain components it should be `.ts`

File: packages/web/utils/meTokens.tsx
Comment (alalonde) [inline packages/web/utils/meTokens.tsx:17]: I believe we have this constant defined elsewhere. If so, use that one

File: packages/web/utils/meTokens.tsx
Comment (alalonde) [inline packages/web/utils/meTokens.tsx:128]: probably shouldn't be using `await` and `.then`

File: packages/web/utils/meTokens.tsx
Comment (alalonde) [inline packages/web/utils/meTokens.tsx:23]: what is the point of this switch statement?

File: packages/web/contracts/FoundryFacet.abi.ts
Comment (alalonde) [inline packages/web/contracts/FoundryFacet.abi.ts:1]: @dysbulic What is the best practice for including these contract definitions? Seems like we should be pulling them in from elsewhere if possible

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (alalonde) [inline packages/web/components/Player/Section/PlayerMeToken.tsx:116]: Ditto, handlePreview itself has a bunch of dependencies

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (alalonde) [inline packages/web/components/Player/Section/PlayerMeToken.tsx:288]: either use an existing function in `utils` or `packages/utils` or move these utility functions there

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (alalonde) [inline packages/web/components/Player/Section/PlayerMeToken.tsx:285]: what about other wallet providers?

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (alalonde) [inline packages/web/components/Player/Section/PlayerMeToken.tsx:540]: I think there is an existing component that does this. See `SwitchNetworkButton`

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (alalonde) [inline packages/web/components/Player/Section/PlayerMeToken.tsx:591]: Please explain why  dependencies are omitted in a comment or fix. Same with the one below

**Review by alalonde — CHANGES_REQUESTED** [review]
Thanks for putting this together!

My main concern is what seems to be improper `useEffect`s throughout.

File: packages/web/utils/meTokens.tsx
Comment (Seroxdesign) [reply packages/web/utils/meTokens.tsx:23]: There will be more collateral tokens, we match them with their ID to query coingecko api

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (Seroxdesign) [reply packages/web/components/Player/Section/PlayerMeToken.tsx:288]: Moved to utils/mathhelpers

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (Seroxdesign) [reply packages/web/components/Player/Section/PlayerMeToken.tsx:285]: Used SwitchNetworkButton (as mentioned in comment below)

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (Seroxdesign) [reply packages/web/components/Player/Section/PlayerMeToken.tsx:540]: Used SwitchNetworkButton

File: packages/web/utils/meTokens.tsx
Comment (Seroxdesign) [reply packages/web/utils/meTokens.tsx]: Resolved by using try/catch blocks, example:

```
   try {
      const tx = await meTokenFoundry
      .calculateMeTokensMinted(meTokenAddress, amount)
      return tx;
    } catch (e) {
      throw e;
    }
 
```

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (Seroxdesign) [reply packages/web/components/Player/Section/PlayerMeToken.tsx:111]: Added all the deps in array

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (Seroxdesign) [reply packages/web/components/Player/Section/PlayerMeToken.tsx:116]: Moved this to a useCallback

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (Seroxdesign) [reply packages/web/components/Player/Section/PlayerMeToken.tsx:591]: Added all deps to dependency array

File: packages/web/components/Player/Section/PlayerMeToken.tsx
Comment (Seroxdesign) [reply packages/web/components/Player/Section/PlayerMeToken.tsx]: Wrapped all functions in useCallback, moved utils out to utils/numbers

File: packages/web/contracts/FoundryFacet.abi.ts
Comment (dysbulic) [reply packages/web/contracts/FoundryFacet.abi.ts:1]: They're coming from Etherscan, so we have to include them in our repo somewhere.

**Review by alalonde — APPROVED** [review]

**Comment by alalonde** [comment]
I would like to see someone from meTokens view [this deployment](https://frontend-pr-1541-mjhnbmqqna-uk.a.run.app/) before we merge this.

Also this needs a rebase still.

**Comment by Seroxdesign** [comment]
> I would like to see someone from meTokens view [this deployment](https://frontend-pr-1541-mjhnbmqqna-uk.a.run.app/) before we merge this.
> 
> Also this needs a rebase still.

I've rebased, waiting on an approval comment from Cbob @metokens

**Pull request closed without merging**
